labcontrol/gui/static/js/addPlateModal.js

```diff
@@ -57,86 +64,95 @@ function insertPlateModalDiv(elementIdToAppendTo = undefined,
-                            specificAddPlateBtnBaseId + plateId +
-                            "' class='btn btn-success btn-circle-small'>" +
-                            "<span class='glyphicon glyphicon-plus'></span>" +
-                            "</button>";
-                    }
-                },
-                {
-                    'targets': -2, // second to last column
-                    'data': null,
-                    'render': function (data, type, row, meta) {
-                        // index 3 in row is the list of studies
-                        return data[3].join('<br/>');
-                    }
-                }
-            ],
-            // per https://datatables.net/reference/option/destroy:
-            // Initialise a new DataTable as usual, but if there is an existing
-            // DataTable which matches the selector, it will be destroyed and
-            // replaced with the new table.
-            'destroy': true,
-            'order': [[0, "desc"]] // order rows in desc order by plate id
+ * This method creates and populates a DataTables object with the input
+ * plateListInfo, and sets it as the contents of the dom object with the
+ * JQuery selector tableSelectorStr (which must already exist in the dom).
+ * tableSelectorStr is the string specifying the JQuery selector for the table
+ * to be populated, e.g. "#searchPlateTable".
+ *
+ * plateListInfo is a nested list of the format
+ * [
+ *   [21, "Test plate 1", "2019-05-09 21:21:15.386036",
+ *       ["Identification of the Microbiomes for Cannabis Soils"]
+ *   ],
+ *   [27, "Test plate 2", "2019-05-09 21:21:15.386036",
+ *       ["Identification of the Microbiomes for Cannabis Soils",
+ *       "Some Other Qiita Study"]
+ *   ]
+ *  ]
+ * ... where the outer list contains one entry for each plate, which in turn
+ * contains four entries: the plate id, the plate external id (e.g., name),
+ * the date the plate was created, and a list of the names of all the Qiita
+ * studies to which ANY of the samples on the plate belong.
+ * specificAddPlateBtnBaseId is the base part--that is, the part that doesn't
+ * actually include the plate id--of the the string id for the button one
+ * clicks to add a particular, specific plate; e.g., "addBtnPlate".
+ */
+function populatePlateTable(
+  plateListInfo,
+  tableSelectorStr,
+  specificAddPlateBtnBaseId
+) {
+  // populate the table of potential plates to add
+  return $(tableSelectorStr).DataTable({
+    data: plateListInfo,
+    columnDefs: [
+      {
+        targets: -1, // last column
+        data: null,
+        render: function(data, type, row, meta) {
+          var plateId = data[0];
+          return (
+            "<button id='" +
+            specificAddPlateBtnBaseId +
+            plateId +
+            "' class='btn btn-success btn-circle-small'>" +
+            "<span class='glyphicon glyphicon-plus'></span>" +
+            "</button>"
+          );
```

Comment:
Note for other readers - the prettifier has encapsulated return's params in an (), and this affected formatting.